### YamlMime:JavaType
uid: "com.azure.core.exception.ClientAuthenticationException"
fullName: "com.azure.core.exception.ClientAuthenticationException"
name: "ClientAuthenticationException"
nameWithType: "ClientAuthenticationException"
summary: "The exception thrown when failed to authenticate the client request with status code of 4XX, typically 401 unauthorized."
inheritances:
- "<xref href=\"java.lang.Object?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"java.lang.Throwable?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"java.lang.Exception?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"java.lang.RuntimeException?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"com.azure.core.exception.AzureException?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"com.azure.core.exception.HttpResponseException?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
inheritedClassMethods:
- classRef: "<xref href=\"com.azure.core.exception.HttpResponseException?alt=com.azure.core.exception.HttpResponseException&text=HttpResponseException\" data-throw-if-not-resolved=\"False\" />"
  methodsRef:
  - "<xref href=\"com.azure.core.exception.HttpResponseException.getResponse()?alt=com.azure.core.exception.HttpResponseException.getResponse&text=getResponse\" data-throw-if-not-resolved=\"False\" />"
  - "<xref href=\"com.azure.core.exception.HttpResponseException.getValue()?alt=com.azure.core.exception.HttpResponseException.getValue&text=getValue\" data-throw-if-not-resolved=\"False\" />"
- classRef: "java.lang.<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html\">Object</a>"
  methodsRef:
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#clone--\">clone</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#equals-java.lang.Object-\">equals</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#finalize--\">finalize</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#getClass--\">getClass</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#hashCode--\">hashCode</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#notify--\">notify</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#notifyAll--\">notifyAll</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#wait--\">wait</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#wait-long-\">wait</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#wait-long-int-\">wait</a>"
- classRef: "java.lang.<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html\">Throwable</a>"
  methodsRef:
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#addSuppressed-java.lang.Throwable-\">addSuppressed</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#fillInStackTrace--\">fillInStackTrace</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#getCause--\">getCause</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#getLocalizedMessage--\">getLocalizedMessage</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#getMessage--\">getMessage</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#getStackTrace--\">getStackTrace</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#getSuppressed--\">getSuppressed</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#initCause-java.lang.Throwable-\">initCause</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#printStackTrace--\">printStackTrace</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#printStackTrace-java.io.PrintStream-\">printStackTrace</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#printStackTrace-java.io.PrintWriter-\">printStackTrace</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#setStackTrace-java.lang.StackTraceElement[]-\">setStackTrace</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#toString--\">toString</a>"
syntax: "public class **ClientAuthenticationException**</br> extends <xref href=\"com.azure.core.exception.HttpResponseException?alt=com.azure.core.exception.HttpResponseException&text=HttpResponseException\" data-throw-if-not-resolved=\"False\" />"
constructors:
- uid: "com.azure.core.exception.ClientAuthenticationException.ClientAuthenticationException(java.lang.String,com.azure.core.http.HttpResponse)"
  fullName: "com.azure.core.exception.ClientAuthenticationException.ClientAuthenticationException(String message, HttpResponse response)"
  name: "ClientAuthenticationException(String message, HttpResponse response)"
  nameWithType: "ClientAuthenticationException.ClientAuthenticationException(String message, HttpResponse response)"
  summary: "Initializes a new instance of the <xref uid=\"com.azure.core.exception.ClientAuthenticationException\" data-throw-if-not-resolved=\"false\" data-raw-source=\"ClientAuthenticationException\"></xref> class."
  parameters:
  - description: "The exception message or the response content if a message is not available."
    name: "message"
    type: "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/String.html\">String</a>"
  - description: "The HTTP response with the authorization failure."
    name: "response"
    type: "<xref href=\"com.azure.core.http.HttpResponse?alt=com.azure.core.http.HttpResponse&text=HttpResponse\" data-throw-if-not-resolved=\"False\" />"
  syntax: "public ClientAuthenticationException(String message, HttpResponse response)"
  desc: "Initializes a new instance of the <xref uid=\"com.azure.core.exception.ClientAuthenticationException\" data-throw-if-not-resolved=\"false\" data-raw-source=\"ClientAuthenticationException\"></xref> class."
- uid: "com.azure.core.exception.ClientAuthenticationException.ClientAuthenticationException(java.lang.String,com.azure.core.http.HttpResponse,java.lang.Object)"
  fullName: "com.azure.core.exception.ClientAuthenticationException.ClientAuthenticationException(String message, HttpResponse response, Object value)"
  name: "ClientAuthenticationException(String message, HttpResponse response, Object value)"
  nameWithType: "ClientAuthenticationException.ClientAuthenticationException(String message, HttpResponse response, Object value)"
  summary: "Initializes a new instance of the <xref uid=\"com.azure.core.exception.ClientAuthenticationException\" data-throw-if-not-resolved=\"false\" data-raw-source=\"ClientAuthenticationException\"></xref> class."
  parameters:
  - description: "The exception message or the response content if a message is not available."
    name: "message"
    type: "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/String.html\">String</a>"
  - description: "The HTTP response with the authorization failure."
    name: "response"
    type: "<xref href=\"com.azure.core.http.HttpResponse?alt=com.azure.core.http.HttpResponse&text=HttpResponse\" data-throw-if-not-resolved=\"False\" />"
  - description: "The deserialized HTTP response value."
    name: "value"
    type: "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html\">Object</a>"
  syntax: "public ClientAuthenticationException(String message, HttpResponse response, Object value)"
  desc: "Initializes a new instance of the <xref uid=\"com.azure.core.exception.ClientAuthenticationException\" data-throw-if-not-resolved=\"false\" data-raw-source=\"ClientAuthenticationException\"></xref> class."
- uid: "com.azure.core.exception.ClientAuthenticationException.ClientAuthenticationException(java.lang.String,com.azure.core.http.HttpResponse,java.lang.Throwable)"
  fullName: "com.azure.core.exception.ClientAuthenticationException.ClientAuthenticationException(String message, HttpResponse response, Throwable cause)"
  name: "ClientAuthenticationException(String message, HttpResponse response, Throwable cause)"
  nameWithType: "ClientAuthenticationException.ClientAuthenticationException(String message, HttpResponse response, Throwable cause)"
  summary: "Initializes a new instance of the <xref uid=\"com.azure.core.exception.ClientAuthenticationException\" data-throw-if-not-resolved=\"false\" data-raw-source=\"ClientAuthenticationException\"></xref> class."
  parameters:
  - description: "The exception message or the response content if a message is not available."
    name: "message"
    type: "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/String.html\">String</a>"
  - description: "The HTTP response with the authorization failure."
    name: "response"
    type: "<xref href=\"com.azure.core.http.HttpResponse?alt=com.azure.core.http.HttpResponse&text=HttpResponse\" data-throw-if-not-resolved=\"False\" />"
  - description: "The <xref uid=\"java.lang.Throwable\" data-throw-if-not-resolved=\"false\" data-raw-source=\"Throwable\"></xref> which caused the creation of this exception."
    name: "cause"
    type: "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html\">Throwable</a>"
  syntax: "public ClientAuthenticationException(String message, HttpResponse response, Throwable cause)"
  desc: "Initializes a new instance of the <xref uid=\"com.azure.core.exception.ClientAuthenticationException\" data-throw-if-not-resolved=\"false\" data-raw-source=\"ClientAuthenticationException\"></xref> class."
type: "class"
desc: "The exception thrown when failed to authenticate the client request with status code of 4XX, typically 401 unauthorized. A runtime exception indicating request authorization failure caused by one of the following scenarios:\n\n *  A client did not send the required authorization credentials to access the requested resource, i.e. Authorization HTTP header is missing in the request\n *  If the request contains the HTTP Authorization header, then the exception indicates that authorization has been refused for the credentials contained in the request header."
metadata: {}
package: "com.azure.core.exception"
artifact: com.azure:azure-core:1.34.0
